src/vmm/src/guest_config/templates/mod.rs

```diff
@@ -57,37 +76,182 @@ pub mod x86_64 {
+        let mut guest_cpuid_map: BTreeMap<CpuidKey, CpuidEntry>;
+
+        // Get the hash map of CPUID data
+        if host_config.cpuid.amd().is_some() {
+            guest_cpuid_map = host_config.cpuid.amd().unwrap().0.clone();
+        } else if host_config.cpuid.intel().is_some() {
+            guest_cpuid_map = host_config.cpuid.intel().unwrap().0.clone();
+        } else {
+            return Err(Error::Cpuid);
+        }
+
+        // Apply CPUID modifiers
+        for leaf_mod in &template.cpuid_modifiers {
+            let cpuid_key = CpuidKey {
+                leaf: leaf_mod.leaf,
+                subleaf: leaf_mod.subleaf,
+            };
+
+            let cpuid_entry_option = guest_cpuid_map.get(&cpuid_key);
+            let mut guest_cpuid_entry = if let Some(entry) = cpuid_entry_option {
+                entry.clone()
+            } else {
+                CpuidEntry::default()
+            };
+            guest_cpuid_entry.flags = leaf_mod.flags;
+
+            let (mut mod_eax, mut mod_ebx, mut mod_ecx, mut mod_edx) = (
+                u64::from(guest_cpuid_entry.result.eax),
+                u64::from(guest_cpuid_entry.result.ebx),
+                u64::from(guest_cpuid_entry.result.ecx),
+                u64::from(guest_cpuid_entry.result.edx),
+            );
+
+            for reg_mod in &leaf_mod.modifiers {
+                match reg_mod.register {
+                    CpuidRegister::Eax => mod_eax = apply_mask(Some(&mod_eax), &reg_mod.mask),
+                    CpuidRegister::Ebx => mod_ebx = apply_mask(Some(&mod_ebx), &reg_mod.mask),
+                    CpuidRegister::Ecx => mod_ecx = apply_mask(Some(&mod_ecx), &reg_mod.mask),
+                    CpuidRegister::Edx => mod_edx = apply_mask(Some(&mod_edx), &reg_mod.mask),
+                }
+            }
+
+            guest_cpuid_entry = CpuidEntry {
+                flags: leaf_mod.flags,
+                result: CpuidRegisters {
+                    eax: mod_eax as u32,
+                    ebx: mod_ebx as u32,
+                    ecx: mod_ecx as u32,
+                    edx: mod_edx as u32,
+                },
+            };
+
+            guest_cpuid_map.insert(cpuid_key, guest_cpuid_entry);
+        }
+
+        // Apply MSR modifiers
+        for modifier in &template.msr_modifiers {
+            guest_msrs_map.insert(
+                modifier.addr,
+                apply_mask(host_config.msrs.get(&modifier.addr), &modifier.mask),
+            );
+        }
+
+        if host_config.cpuid.amd().is_some() {
+            Ok(X86_64CpuConfiguration {
+                cpuid: Cpuid::Amd(AmdCpuid(guest_cpuid_map)),
+                msrs: guest_msrs_map,
+            })
+        } else if host_config.cpuid.intel().is_some() {
+            Ok(X86_64CpuConfiguration {
+                cpuid: Cpuid::Intel(IntelCpuid(guest_cpuid_map)),
+                msrs: guest_msrs_map,
+            })
+        } else {
+            Err(Error::Internal)
```

Comment:
can we be more descriptive here?

Reply:
Strange, this error was removed already. Can you refresh to confirm?

Reply:
Indeed, it seems to have gone. Probably in later commits.